app/(tabs)/(home)/index.tsx

```diff
@@ -166,61 +164,78 @@ export default function HomeScreen() {
-          if (lat < minLat) minLat = lat;
-          if (lat > maxLat) maxLat = lat;
-          if (lng < minLng) minLng = lng;
-          if (lng > maxLng) maxLng = lng;
-        }
-      });
+    if (!geoJSON) {
+      throw new Error("Keine Route importiert!");
+    }
 
-      const bounds = {
-        ne: [maxLng, maxLat],
-        sw: [minLng, minLat],
-      };
+    setUserCentered(false);
+    // Find the outermost coordinates
+    let minLat = Infinity,
+      maxLat = -Infinity,
+      minLng = Infinity,
+      maxLng = -Infinity;
+    geoJSON.features.forEach((feature) => {
+      if (feature.geometry.type === "LineString") {
+        (feature.geometry as GeoJSON.LineString).coordinates.forEach(
+          ([lng, lat]) => {
+            if (lat < minLat) minLat = lat;
+            if (lat > maxLat) maxLat = lat;
+            if (lng < minLng) minLng = lng;
+            if (lng > maxLng) maxLng = lng;
+          },
+        );
+      } else if (feature.geometry.type === "Point") {
+        const [lng, lat] = (feature.geometry as GeoJSON.Point).coordinates;
+        if (lat < minLat) minLat = lat;
+        if (lat > maxLat) maxLat = lat;
+        if (lng < minLng) minLng = lng;
+        if (lng > maxLng) maxLng = lng;
+      }
+    });
 
-      camera.current?.setCamera({
-        bounds: bounds,
-        padding: {
-          paddingLeft: 30,
-          paddingRight: 30,
-          paddingTop: 30,
-          paddingBottom: 150,
-        },
-        animationDuration: 2000,
-        heading: 0,
-        animationMode: "flyTo",
-      });
-    }
+    const bounds = {
+      ne: [maxLng, maxLat],
+      sw: [minLng, minLat],
+    };
+
+    camera.current?.setCamera({
+      bounds: bounds,
+      padding: {
+        paddingLeft: 30,
+        paddingRight: 30,
+        paddingTop: 30,
+        paddingBottom: 150,
+      },
+      animationDuration: 2000,
+      heading: 0,
+      animationMode: "flyTo",
+    });
   };
 
-  const RouteIcon = (props?: Partial<ImageProps>): IconElement => (
-    <Icon
-      {...props}
-      name="route"
-      style={[props?.style, { height: 40, width: "auto" }]}
-    />
+  const CenterButton = (props?: Partial<ImageProps>) => (
+    <TouchableOpacity
+      style={styles.centerButton}
+      onPress={() => setUserCentered(true)}
+    >
+      <Icon
+        {...props}
+        name="location-crosshairs"
+        style={[props?.style, { height: 23 }]}
+      />
+    </TouchableOpacity>
   );
 
-  const renderRouteAction = (): React.ReactElement => (
-    <TopNavigationAction icon={RouteIcon} onPress={showRoute} hitSlop={15} />
+  const RouteButton = (props?: Partial<ImageProps>) => (
+    <TouchableOpacity
+      style={styles.routeButton}
+      onPress={async () => {
+        setUserCentered(false);
+        await sleepAsync(100);
+        showRoute();
+      }}
+    >
+      <Icon {...props} name="route" style={[props?.style, { height: 23 }]} />
```

Comment:
```suggestion
      <Icon {...props} name="route" style={[props?.style, { height: 22 }]} />
```
Icon ist unten leicht abgeschnitten